pkg/asset/installconfig/azure/validation.go

```diff
@@ -216,11 +234,20 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 				ultraSSDCapability = defaultUltraSSDCapability
 			}
 			if vmNetworkingType == "" {
-				vmNetworkingType = defaultVMNetworkingType
+				capabilities, _ = client.GetVMCapabilities(context.TODO(), instanceType, ic.Azure.Region)
```

Comment:
This is not correct. You need to fetch the capabilities whether `vmNetworkingType` was supplied or not. And you definitely need to check for the `err` value as you did in the controlPlane section. The `instanceType` for the compute nodes might be different from the one for the controlPlane nodes, so you effectively have 2 sets of capabilities.
```
capabilities, err = client.GetVMCapabilities(context.TODO(), instanceType, ic.Azure.Region)
if err != nil {
	fp := field.NewPath("compute", "platform", "azure")
	return append(allErrs, field.Invalid(fp.Child("type"), instanceType, err.Error()))
}
```

Reply:
@r4f4 I agree. This is why (from our slack discussion) I said that I made an attempt at this but didn't like how it was turning out. This is due to the previous comment I discussed above about setting default variables in the middle of the validation functions. The issue is when a user doesn't define the control plane or the compute or any combination of these we will get different combination of validation or none at all.

The getVmCapabilities should occur as one of the first things that happens when doing the validation if the compute || the control plane objects aren't nil.